style/layer/highway_shield.js

```diff
@@ -1,49 +1,48 @@
 "use strict";
 
-export const interstate = {
-  id: "highway-shield-us-interstate",
-  type: "symbol",
-  paint: {
-    "text-color": "rgba(255, 255, 255, 1)",
-  },
-  filter: [
-    "all",
-    ["<=", "ref_length", 6],
-    ["==", "$type", "LineString"],
-    ["in", "network", "us-interstate"],
-  ],
-  layout: {
-    "icon-size": 2.0,
-    "text-font": ["Open Sans Bold"],
-    "text-size": 10.5,
-    "text-anchor": "center",
-    "text-field": "{ref}",
-    "text-offset": [0, 0],
-    "icon-image": "us_interstate",
-    "icon-text-fit": "both",
-    "icon-padding": 0,
-    "icon-text-fit-padding": [2.5, 0.6, 0, 0.5],
-    "symbol-spacing": {
-      stops: [
-        [7, 150],
-        [13, 300],
-        [16, 800],
-      ],
-    },
+import * as Util from "../js/util.js";
+
+function routeConcurrency(num) {
+  return [
+    "case",
+    ["!=", ["get", "route_" + num], null],
+    ["image", ["concat", "shield_", ["get", "route_" + num]]],
+    ["literal", ""],
+  ];
+}
+
+let shieldTextField = ["format"];
+for (var i = 1; i <= 5; i++) {
+  shieldTextField.push(routeConcurrency(i));
+  shieldTextField.push(" ");
+}
+shieldTextField.push(routeConcurrency(6));
+
+let shieldLayout = {
+  "text-rotate-to-line": false,
+  "text-font": ["Metropolis Light"],
+  "text-field": shieldTextField,
+  "text-anchor": "center",
+  "symbol-placement": "line",
+};
 
-    "symbol-placement": {
-      base: 1,
-      stops: [
-        [7, "point"],
-        [7, "line"],
-        [8, "line"],
-      ],
-    },
-    "symbol-avoid-edges": true,
-    "icon-rotation-alignment": "viewport",
-    "text-rotation-alignment": "viewport",
-  },
+let baseShield = {
+  type: "symbol",
+  layout: shieldLayout,
   source: "openmaptiles",
-  minzoom: 7,
+  metadata: {},
   "source-layer": "transportation_name",
 };
+
+function shieldLayer(hwyClass, minzoom) {
+  var layer = Util.cp(baseShield);
+  layer.filter = ["all", ["==", "class", hwyClass], ["has", "route_1"]];
```

Comment:
Some features have route_2 but not route_1, so this filter is hiding them.

Reply:
Oh, I didn't think that would be possible 🤔 .  Since we're going to be getting rid of the spaces we can change the logic to any route_X which should be properly inclusive.

Reply:
Resolved by @jleedev